packages/addon/lua/gelly/api/mods/mod-repository.lua

```diff
@@ -0,0 +1,46 @@
+local MODS_TABLE_NAME = "gelly_mods"
+
+---@alias gellyx.mods.ModMetadata {enabled: boolean}
+
+---@class gellyx.mods.ModRepository
+--- Encapsulates the database operations for mods.
+local repository = {}
+
+function repository.initialize()
+	if not sql.TableExists(MODS_TABLE_NAME) then
+		sql.Query("CREATE TABLE " .. MODS_TABLE_NAME .. " (mod_id TEXT PRIMARY KEY, enabled BOOLEAN)")
+	end
+end
+
+--- Upserts metadata for a mod ID.
+---@param modId string
+---@param metadata gellyx.mods.ModMetadata
+---@return nil
+function repository.upsertMetadataForModId(modId, metadata)
+	local query = ("INSERT INTO %s (mod_id, enabled) VALUES (%s, %s) ON CONFLICT (mod_id) DO UPDATE SET enabled = %s")
```

Comment:
This upsert doesn't really seem sound, especially cause we're manually constructing the query based on the table's convention. We may need to introduce a new module which can construct queries based on the actual structure of the table, but I don't see this as a blocker.

Reply:
Agreed, will probably raise a ticket for this, although the priority is 100% going to be low.